### PR TITLE
Enable Dora metastore cache

### PR DESCRIPTION
**Pull request by huanghua78**
### What changes are proposed in this pull request?

Store File Metadata in persistent DoraMetaStore which is stored in RocksDB.

### Why are the changes needed?

We want to have File Metadata cache in memory and on persistent storage.
This is to enable the persistent cache.

### Does this PR introduce any user facing changes?

N/A

### CODE REVIEW

**Comment by huanghua78** [comment]
Hello Reviewers, please note, the base branch is on https://github.com/Alluxio/alluxio/pull/16847
When you do reviewing, please pay your attentions to the last commit.
All other commits are in https://github.com/Alluxio/alluxio/pull/16847

File: core/server/worker/src/main/java/alluxio/worker/dora/PagedDoraWorker.java [hunk 1]
```diff
@@ -119,6 +121,8 @@ public void start(WorkerNetAddress address) throws IOException {
             ImmutableMap.of(),
             ImmutableMap.of(),
             Configuration.getConfiguration(Scope.WORKER));
+        String dbDir = Configuration.getString(PropertyKey.DORA_WORKER_METASTORE_ROCKSDB_DIR);
+        mMetaStore = new RocksDBDoraMetaStore(dbDir + ".worker." + mWorkerId);
```
Comment (dbw9580) [inline core/server/worker/src/main/java/alluxio/worker/dora/PagedDoraWorker.java:125]: close the metastore when the worker is closed.

**Review by dbw9580 — APPROVED** [review]
LGTM except a small issue

File: core/server/worker/src/main/java/alluxio/worker/dora/PagedDoraWorker.java (hunk 1 above)
Comment (huanghua78) [reply core/server/worker/src/main/java/alluxio/worker/dora/PagedDoraWorker.java:125]: Oh, yes!

Comment (huanghua78) [reply core/server/worker/src/main/java/alluxio/worker/dora/PagedDoraWorker.java:125]: fixed.

File: core/server/worker/src/main/java/alluxio/worker/grpc/DoraWorkerClientServiceHandler.java [hunk 2]
```diff
@@ -108,30 +118,67 @@ public StreamObserver<ReadRequest> readBlock(StreamObserver<ReadResponse> respon
...
+      UfsFileStatus status = mUfsFileStatusCache.getIfPresent(ufsFullPath);
+      if (status == null) {
+        // The requested FileStatus is not present in memory cache.
+        // Let's try to query local persistent DoraMetaStore.
+        DoraMetaStore doraMetaStore = ((PagedDoraWorker) mWorker).getMetaStore();
+        Optional<DoraMeta.FileStatus> fs = doraMetaStore.getDoraMeta(ufsFullPath);
+        if (fs.isPresent()) {
+          // Found in persistent DoraMetaStore
+          fi = fs.get().getFileInfo();
+          String contentHash = UnderFileSystemUtils.approximateContentHash(fi.getLength(),
```
Comment (jja725) [inline core/server/worker/src/main/java/alluxio/worker/grpc/DoraWorkerClientServiceHandler.java:137]: I think it's fine for now but in the future I may need accurate contentHash for dstCP

**Review by jja725 — APPROVED** [review]
LGTM

File: core/server/worker/src/main/java/alluxio/worker/grpc/DoraWorkerClientServiceHandler.java (hunk 2 above)
Comment (huanghua78) [reply core/server/worker/src/main/java/alluxio/worker/grpc/DoraWorkerClientServiceHandler.java:137]: OK.

**Comment by huanghua78** [comment]
git rebased to latest base branch.

**Comment by beinan** [comment]
@huanghua78 sorry, I should merge this one first.  could you rebase again?

**Comment by huanghua78** [comment]
> @huanghua78 sorry, I should merge this one first. could you rebase again?

OK. Let me do it.

**Comment by huanghua78** [comment]
> > @huanghua78 sorry, I should merge this one first. could you rebase again?
> 
> OK. Let me do it.

Rebases. Let'e wait for the test results.

**Comment by huanghua78** [comment]
rebased to the latest `dora` branch.

**Comment by beinan** [comment]
alluxio-bot, merge this please.

**Pull request merged**